src/impls.rs

```diff
@@ -90,6 +91,15 @@ mod bstring {
         }
     }
 
+    impl FromStr for BString {
```

Comment:
you should be able to impl this for `&BStr` too.

Reply:
I believe that should not be done, as `FromStr` expects the output to be owned. In that case you probably have to use `From` instead. I tested and I cannot specify the same lifetime for the `&str` and the `&BStr`.